api/app/utils/db.ts

```diff
@@ -98,4 +99,33 @@ export class ConnectionWrapper {
     }
 
 
-}
\ No newline at end of file
+}
+
+
+const LIMITED = Symbol('connection_limited')
+
+declare module 'mysql2/promise' {
+    interface Connection {
+        [LIMITED]?: true
+    }
+}
+
+export function decorateLimitedPool(pool: Pool, limits: string[]) {
```

Comment:
Can we use this as Typescript decorators for a TiDBQueryExecutor instance?

https://www.typescriptlang.org/docs/handbook/decorators.html

Reply:
No, the declaration of type `Pool` is an interface not class.

Reply:
The `connection` event will not block the `getConnection` procedure, which would lead to an incomplete initialization state.

Reply:
👍 Got it, good catch!